Makefile.am

```diff
@@ -1,5 +1,7 @@
 DIST_SUBDIRS = yajl
+if HAVE_EMBEDDED_YAJL
```

Comment:
A small question: Is this check necessary for this PR ? Cause we a just configured `DIST_SUBDIRS` to the same value. I think we should be fine even with dropping this check right ?

Reply:
good question.  @ronalabraham why do you need to add this check?

Reply:
After looking into this further, I am OK with reverting this. The intention was to allow `libocispec` to build and install *without* having to pull down the `yajl` submodule. In our integration build, we would do the rough equivalent of:
```sh
git submodule update --init --recursive image-spec runtime-spec
autoreconf --force --install --verbose
./configure --disable-embedded-yajl --enable-libocispec-install
make
sudo make install
```
I've decided that this is unnecessarily complicated and brittle. It would also require some additional changes to [`configure.ac`](https://github.com/containers/libocispec/blob/65bdde6f82b5ce8cddc316b8f93e30bc1c35e333/configure.ac#L15-L16) to protect these lines with `HAVE_EMBEDDED_YAJL`:
```
    AM_EXTRA_RECURSIVE_TARGETS([yajl])
    AC_CONFIG_SUBDIRS([yajl])
```
We'll just go ahead and pull `yajl` down when we build and install.